### PR TITLE
Using apimachinery rand and changed length to 20

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue

fixes https://github.com/flyteorg/flyte/issues/2159

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by katrogan** [comment]
is this ready for review? should we update GetExecutionName like the referenced issue suggests?

File: pkg/common/executions.go [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
-// In kubernetes, resource names must comply with this regex: '[a-z]([-a-z0-9]*[a-z0-9])?'
-const AllowedExecutionIDStartCharStr = "abcdefghijklmnopqrstuvwxyz"
-const AllowedExecutionIDStr = "abcdefghijklmnopqrstuvwxyz1234567890"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
 
-var AllowedExecutionIDStartChars = []rune(AllowedExecutionIDStartCharStr)
-var AllowedExecutionIDChars = []rune(AllowedExecutionIDStr)
+const ExecutionIDLength = 20
+const ExecutionStringFormat = "f%s"
```
Comment (katrogan) [inline pkg/common/executions.go:12]: thoughts on having the start char be something different than 'f' so we can easily distinguish between propeller-launched executions vs. admin-generated names?

**Review by katrogan — APPROVED** [review]

File: pkg/common/executions.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply pkg/common/executions.go]: Make sense. Will make change in flytectl too so that we can distinguish those aswell.

Comment (katrogan) [reply pkg/common/executions.go]: awesome, thank you!

**Review by katrogan — APPROVED** [review]

**Pull request merged**